spec/furniture/marketplace/product_spec.rb

```diff
@@ -18,4 +18,50 @@
       expect(product.price_cents).to be(2000)
     end
   end
+
+  describe "#discardable?" do
+    subject(:delivery_area) { build(:marketplace_delivery_area) }
+
+    it { is_expected.not_to be_discardable }
+
+    context "when the delivery area is persisted" do
+      subject(:delivery_area) { create(:marketplace_delivery_area) }
+
+      it { is_expected.to be_discardable }
+
+      context "when the delivery area is discarded already" do
+        subject(:delivery_area) { create(:marketplace_delivery_area, :discarded) }
+
+        it { is_expected.not_to be_discardable }
+      end
+    end
+  end
+
+  describe "#destroyable?" do
+    subject(:product) { build(:marketplace_product) }
+
+    it { is_expected.not_to be_destroyable }
+
+    context "when a delivery area is persisted" do
+      subject(:product) { create(:marketplace_product) }
+
+      it { is_expected.not_to be_destroyable }
+
+      context "when the delivery area is discarded" do
+        subject(:product) { create(:marketplace_product, :discarded) }
+
+        it { is_expected.to be_destroyable }
+
+        context "when the delivery area has orders" do
+          subject(:product) { create(:marketplace_product, :discarded) }
+
+          let(:order) { create(:marketplace_order) }
+
+          before { order.products << product }
+
+          it { is_expected.not_to be_destroyable }
+        end
+      end
+    end
+  end
```

Comment:
Might be worth having a test case that verifies that when a product that is in a pre-checkout cart is discarded, it gets removed from those carts.

Reply:
You have a test for that in the system spec already. 👍🏼

Reply:
Yea, I think I want to push the discard-removal checks down into the model specs; I just wasn't sure where to put it when I started and I started with the system tests.